### PR TITLE
Remove uPayload

### PR DESCRIPTION
**Pull request by stevenhartley**
The following change removes the unused UPayload protobuf object and places the format in UAttributes.

#144

### CODE REVIEW

File: up-core-api/uprotocol/uattributes.proto [hunk 1]
```diff
@@ -118,3 +118,34 @@ enum UPriority {
     // Network control such as Safety Critical
     UPRIORITY_CS6 = 7 [(uprotocol.ce_name) = "CS6"];
 }
+
+// The format for the data stored in the UMessage.
+enum UDataContentType {
```
Comment (gregmedd) [inline up-core-api/uprotocol/uattributes.proto:123]: Shouldn't a field of this type be added to `UAttributes`?

**Comment by gregmedd** [comment]
This looks like an otherwise minor change to the structure of a message that has fairly large implications for the APIs we are building. For example, this will require non-trivial changes to the `up-cpp` API update we just reviewed and approved yesterday as it completely changes the interaction model for message data.

It also removes the possibility of extending the `UPayload` type in the future should it be necessary to support more advanced features, although I can see that the intent is that those would be represented as new data types so maybe extensibility doesn't matter so much.

Are we really sure it is worth making this change? If nothing else, it will further delay the `up-cpp` API updates while we make this change.

File: up-core-api/uprotocol/uattributes.proto (hunk 1 above)
Comment (stevenhartley) [reply up-core-api/uprotocol/uattributes.proto:126]: Done, rooky mistake :-)

File: basics/uattributes.adoc [hunk 2]
```diff
@@ -78,6 +78,11 @@ stem:[t_{current} > t_{id} + ttl]
 |no
 |A tracing identifier to use for correlating messages across the system. Intended to be compatible with https://github.com/cloudevents/spec/blob/main/cloudevents/extensions/distributed-tracing.md[CloudEvents distributed tracing extension].
 
+|`datacontentytpe`
```
Comment (PLeVasseur) [inline basics/uattributes.adoc:81]:
```suggestion
|`datacontenttype`
```
?

File: up-core-api/uprotocol/uattributes.proto (hunk 1 above)
Comment (PLeVasseur) [inline up-core-api/uprotocol/uattributes.proto:126]: We need to put a `UDataContentType` inside of `UAttributes`, right?

File: basics/upayloadformat.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+SPDX-FileType: DOCUMENTATION
+SPDX-License-Identifier: Apache-2.0
+----
+
+== Overview
+
+The following documents the different data content types (data encoding formats) that are supported by uProtocol.
+Data content types are declared in link:../up-core-api/uprotocol/uattributes.proto[uattributes.proto].
+
+== UDataContentType
```
Comment (sophokles73) [inline basics/upayloadformat.adoc:27]: UPayloadFormat?

File: up-core-api/uprotocol/uattributes.proto [hunk 4]
```diff
@@ -118,3 +121,34 @@ enum UPriority {
...
+
+    // RAW (binary) format
+    UPAYLOAD_FORMAT_RAW = 6 [ (uprotocol.mime_type) = "application/octet-stream" ];
+
+    // Text format
+    UPAYLOAD_FORMAT_TEXT = 7 [ (uprotocol.mime_type) = "text/plain" ];
+
+    // Shared memory format
+    UPAYLOAD_FORMAT_SHM = 8 [ (uprotocol.mime_type) = "application/x-shm" ];
+}
```
Comment (sophokles73) [inline up-core-api/uprotocol/uattributes.proto:154]: missing EOL

File: basics/upayloadformat.adoc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+|Name |Mime Type |Data Description
+
+|*WRAPPED_IN_ANY* | `application/x-protobuf` | Serialized `google.protobuf.Any` message.
+|*PROTOBUF* |`application/protobuf` |Protobuf serialized `google.protobuf.Message`
+|*JSON* |`application/json` |JSON serialization format
+|*SOME_IP* |`application/x-someip` |Basic SOME/IP serialization format
+|*SOME_IP_TLV* |`application/x-someip_tlv` |SOME/IP TLV format
+|*RAW* |`application/octet-stream` |RAW (binary) format
+|*TEXT* |`text/plain` |Text format
+| *SHM* | `application/x-shm` | Shared memory format
```
Comment (sophokles73) [inline basics/upayloadformat.adoc:40]: I would propose to use the definitions from the original upayload.adoc file and amend it with `SHM`

File: basics/upayloadformat.adoc [hunk 6]
```diff
@@ -0,0 +1,42 @@
+= Data Content Type
```
Comment (sophokles73) [inline basics/upayloadformat.adoc:1]: Payload Format?

File: up-core-api/uprotocol/umessage.proto [hunk 7]
```diff
@@ -16,19 +16,18 @@ syntax = "proto3";
...
+// It contains a header (UAttributes), and payload and is a way of representing a
 // message that would be sent between two uEs.
 message UMessage {
     // uProtocol mandatory and optional attributes
     UAttributes attributes = 1;
 
-    // Optional message payload containing the data to be sent
-    UPayload payload = 2;
+    // Optional message payload
+    optional bytes payload = 2;
```
Comment (sophokles73) [inline up-core-api/uprotocol/umessage.proto:32]: missing EOL at end of file

File: basics/upayloadformat.adoc (hunk 3 above)
Comment (stevenhartley) [reply basics/upayloadformat.adoc:27]: resolved

File: up-core-api/uprotocol/uattributes.proto (hunk 4 above)
Comment (stevenhartley) [reply up-core-api/uprotocol/uattributes.proto:154]: done

File: basics/upayloadformat.adoc (hunk 5 above)
Comment (stevenhartley) [reply basics/upayloadformat.adoc:40]: I reverted to the old and then renamed the file.

File: basics/upayloadformat.adoc (hunk 6 above)
Comment (stevenhartley) [reply basics/upayloadformat.adoc:1]: done.

File: up-core-api/uprotocol/umessage.proto (hunk 7 above)
Comment (stevenhartley) [reply up-core-api/uprotocol/umessage.proto:32]: done.

File: basics/README.adoc [hunk 8]
```diff
@@ -28,7 +28,6 @@ The following are the foundational requirements for uProtocol specifications:
 * *link:qos.adoc[Quality of Service & Priority (UPriority)]*
 * *link:uattributes.adoc[uProtocol Attributes (UAttributes)]*
 * *link:umessage.adoc[uProtocol Message (UMessage)]*
-* *link:upayload.adoc[uProtocol Payload (UPayload)]*
```
Comment (sophokles73) [inline basics/README.adoc:31]: maybe replace this with the link to upayloadformat.adoc?

Comment (stevenhartley) [reply basics/README.adoc:31]: done

**Review by PLeVasseur — APPROVED** [review]
LGTM

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**